Fix startup tests

jclarkeSTFC opened the pull request:
Since we switched to `pip install --editable` instead of calling `setup.py` directly (#37775), we no longer get an executable file in the build output, so this test needed adjusting. It was also attempting to do a for loop and returning at the first entry so we fixed that.

On Linux it was trying to test `mantidworkbench` as an entry point but:
- Because it was previously returning at the first item in the loop, this was never actually tested.
- The `mantidworkbench` entry point is created by `conda-build`, not when you do a dev build, so this wouldn't have worked previously anyway.

### To test:

https://builds.mantidproject.org/job/build_packages_from_branch/876/

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jclarkeSTFC committed:
The .exe file is no longer produced in a dev build

Since we switched to pip install --editable instead of calling
setup.py directly, we no longer get an executable file in the build
output, so this test needed adjusting. It was also attempting to do a
for loop and returning at the first entry so we fixed that.

Co-authored-by: Jonathan Haigh <jonathan.haigh@stfc.ac.uk>

sf1919 commented:
This will need a build packages from branch run as the failing systems tests don't run on Windows or OSX as part of normal PR testing

jclarkeSTFC committed:
Remove mantidworkbench entry point test

This was never tested previously, and the dev build does not create a
mantidworkbench entry point, conda-build does that, hence I've deleted
it.

jclarkeSTFC committed:
VS debug command fix

The startup script isn't going to exist any more, so change the command
used for debugging workbench.

jhaigh0 reviewed (commented):
System tests passing on all operating systems 👍

jhaigh0 reviewed (approved):
mantid launching from both vs and pycharm
